src/dtclient/activegate_tenant_info.go

```diff
@@ -0,0 +1,58 @@
+package dtclient
+
+import (
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+type ActiveGateTenantInfo struct {
+	UUID      string `json:"tenantUUID"`
+	Token     string `json:"tenantToken"`
+	Endpoints string `json:"communicationEndpoints"`
+}
+
+func (dtc *dynatraceClient) GetActiveGateTenantInfo() (*ActiveGateTenantInfo, error) {
+	response, err := dtc.makeRequest(
+		dtc.getActiveGateConnectionInfoUrl(),
+		dynatracePaaSToken,
+	)
+
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	defer func() {
+		err := response.Body.Close()
+		if err != nil {
+			log.Error(err, err.Error())
+		}
+	}()
+
+	data, err := dtc.getServerResponseData(response)
+	if err != nil {
+		return nil, dtc.handleErrorResponseFromAPI(data, response.StatusCode)
+	}
+
+	tenantInfo, err := dtc.readResponseForActiveGateTenantInfo(data)
+	if err != nil {
+		log.Error(err, err.Error())
+		return nil, err
+	}
+	if len(tenantInfo.Endpoints) == 0 {
+		log.Info("tenant has no endpoints")
+	}
+
+	return tenantInfo, nil
```

Comment:
I'm missing `tenantInfo.UUID` validation here, what if it's empty or whitespace-only?

Reply:
`dtclient` should be refactored - for example generated with swagger. I don't want to manually provide sophisticated logic here because it requires maintenance.
If `tenantInfo.UUID` is empty, or whitespace-only, we can do nothing except printing warning. I see no value to user here because pod creation is failing anyway with obvious reason.